astropy/coordinates/earth.py

```diff
@@ -620,18 +622,57 @@ def get_gcrs_posvel(self, obstime):
             The GCRS velocity of the object
         """
         # do this here to prevent a series of complicated circular imports
-        from .builtin_frames import GCRS
+        from .builtin_frames import CIRS, GCRS
+        from .builtin_frames.intermediate_rotation_transforms import gcrs_to_cirs_mat
 
         itrs = self.get_itrs(obstime)
+        cirs_frame = CIRS(obstime=obstime)
         geocentric_frame = GCRS(obstime=obstime)
+
         # GCRS position
         obsgeoloc = itrs.transform_to(geocentric_frame).cartesian
-        vel_x = -OMEGA_EARTH * obsgeoloc.y
-        vel_y = OMEGA_EARTH * obsgeoloc.x
-        vel_z = 0. * vel_x.unit
-        obsgeovel = CartesianRepresentation(vel_x, vel_y, vel_z)
+
+        # CIRS velocity
+        rotation_vector = CartesianRepresentation(0/u.s, 0/u.s, OMEGA_EARTH)
```

Comment:
Now that we have velocity support in frames at least, it should be possible to not do anything at all, except to add zero velocity to the ITRS position...  PR soon...